src/modules/CurrencyModule.cpp

```diff
@@ -1,46 +1,350 @@
+		
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+		if (args.first() != prefix + "setcurrenname")
+			return;
+
+		if (args.size() > 2) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+
+		if (args.size() == 2) {
+			currencyName = args.at(1);
+			client.createMessage(message.channelId(), "**Currency Name set to** " + currencyName);
+		}
+
+		});
+
+	RegisterCommand(Commands::CURRENCY_SET_SYMBOL, "setcurrensymb", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) {
+
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+		if (args.first() != prefix + "setcurrensymb")
+			return;
+
+		if (args.size() > 2) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+
+		if (args.size() == 2) {
+			currencySymbol = args.at(1);
+			client.createMessage(message.channelId(), "**Currency Symbol set to** " + currencySymbol);
+		}
+
+		});
 }
 
-void CurrencyModule::StatusCommand(QString& result, snowflake_t guild, snowflake_t user)
-{
-	result += "$ " + QString::number(m_settings[user].currency) + "\n";
-	result += "\n";
+void CurrencyModule::OnMessage(Discord::Client& client, const Discord::Message& message) {
+
+	if (!isRandomGiveAwayDone) {
+		static int pingCount = 0;
+		if (pingCount == 0) {
+			bool outcome = randp(random_engine);
+			if (outcome) {
+				client.createMessage(giveawayChannelId, "@everyone **FREEBIE** available now! Go `!claim` some juicy coins!");
+				allowGiveaway = true;
+				pingCount++;
+			}
+		}
+	}
+
+	if (selfGambleData.gamble && !message.author().bot() && message.channelId() == selfGambleData.channelId) {
+		if (message.author().id() == selfGambleData.userId) {
+
+			//Check if the message is a number
+			QRegExp re("\\d*");
+			if (!re.exactMatch(message.content()))
+				return;
+
+			GuildSetting* setting = &GuildSettings::GetGuildSetting(selfGambleData.guildId);
+			if (message.content().toInt() > 5 || message.content().toInt() < 0) {
+
+				client.createMessage(message.channelId(), "**Your guess is out of range!** \nTry a number between 0 and 5 (inclusive)");
+				return;
+
+			}
+			if (message.content().toInt() == selfGambleData.randNum) {
+				m_settings[setting->id].currency += 50;
+				client.createMessage(message.channelId(), "**You guessed CORRECTLY!**\n(**50" + currencySymbol  + "** have been added to your wallet!)");
+			}
+			else {
+				m_settings[setting->id].currency -= 10;
+				client.createMessage(message.channelId(), "**Better Luck next time!**\n*(psst! I took **10CC** from your wallet for my time...)*");
```

Comment:
Here it says "10CC" but in other places it says "50"+currencySymbol. I assume that's what you want to do here too.

Reply:
Oh I might've missed that, thanks!